perfmetrics/scripts/bigquery/bigquery.py

```diff
@@ -0,0 +1,206 @@
+import argparse
+from google.cloud import bigquery
+import sys
+
+PROJECT_ID = 'gcs-fuse-test'
+DATASET_ID = 'performance_metrics'
+CONFIGURATION_TABLE_ID = 'experiment_configuration'
+FIO_TABLE_ID = 'fio_metrics'
+VM_TABLE_ID = 'vm_metrics'
+LS_TABLE_ID = 'ls_metrics'
+
+# Construct a BigQuery client object.
+client = bigquery.Client()
+
+class BigQuery():
+
+  def insert_config_and_get_config_id(self, gcsfuse_flags, branch, end_date) -> int:
+
+    """Gets the experiment configuration and checks if it is already present in the BigQuery tables,
+       If not present: New configuration will be inserted, and it's configuration id will be returned
+       If present: Configuration id will be fetched and returned
+
+    Args:
+      gcsfuse_flags: GCSFuse flags for mounting the test buckets.
+      branch: GCSFuse repo branch to be used for building GCSFuse.
+      end_date: Date upto when tests are run.
+
+    Return:
+      Configuration id of the experiment in the BigQuery tables
+    """
+    dataset_ref = None
+    try:
+      dataset_ref = client.dataset(DATASET_ID)
+    except:
+      dataset_ref = client.create_dataset(DATASET_ID)
+
+    # Query for creating experiment_configuration table if it does not exist
+    query_create_table_experiment_configuration = """
+        CREATE TABLE IF NOT EXISTS {}.{}.{}(
+          configuration_id INT64,
+          gcsfuse_flags STRING,
+          branch STRING,
+          end_date TIMESTAMP,
+          PRIMARY KEY (configuration_id) NOT ENFORCED
+        ) OPTIONS (description = 'Table for storing Job Configurations and respective VM instance name on which the job was run');
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID)
+
+    # API Request to create experiment_configuration if it does not exist
+    results = client.query(query_create_table_experiment_configuration)
+    print(results)
+
+    query_get_configuration_id = """
+      SELECT configuration_id
+      FROM `{}.{}.{}`
+      WHERE gcsfuse_flags = '{}'
+      AND branch = '{}'
+      AND end_date = '{}'
+    """.format(PROJECT_ID, DATASET_ID, CONFIGURATION_TABLE_ID, gcsfuse_flags, branch, end_date)
+
+    config_id = None
+    exists = False
+    query_job = client.query(query_get_configuration_id)
+    for row in query_job:
+      config_id = row['configuration_id']
+      exists = True
+    if not exists:
+      table_ref = dataset_ref.table(CONFIGURATION_TABLE_ID)
+      table = client.get_table(table_ref)
+      row_count = table.num_rows
+      config_id = row_count + 1
+      rows_to_insert = [(config_id, gcsfuse_flags, branch, end_date)]
+      errors = client.insert_rows(table, rows_to_insert)
```

Comment:
Why errors? We need to do proper error handling here. In case of error, we can mark failure and exit.

Reply:
I should change the variable name. The error handling is done above by checking if the configuration already exists. If it does not exist then only the insertion is done. I will change the variable name to something like results